### PR TITLE
MM-46539 : E2E | Webapp |  Fix cypress tests | Self hosted pricing modal

### PR DESCRIPTION
**Pull request by yasserfaraazkhan**
Fixes tests

- self_hosted_pricing_modal_spec.js
- subscriptions_free_trial_spec.js
- company_information_free_trial_spec.js
- payment_free_trial_spec.js.js
- cloud_pricing_modal_spec.js

#### Ticket Link
  Fixes https://mattermost.atlassian.net/browse/MM-46539

```release-note
NONE
```

### CODE REVIEW

**Comment by AGMETEOR** [comment]
Hi @yasserfaraazkhan lmk when it's out of drafts

**Review by nickmisasi — COMMENTED** [review]
Just commenting so review can be re-requested when out of draft

File: e2e/cypress/tests/integration/enterprise/cloud/billing/cloud_pricing_modal_spec.js [hunk 1]
```diff
@@ -227,27 +219,27 @@ describe('Pricing modal', () => {
...
 
-        // *Check that starter card Downgrade button is disabled
+        // * Check that starter card Downgrade button is disabled
         cy.get('#pricingModal').get('#starter').get('#starter_action').should('be.disabled').contains('Downgrade');
 
-        // *Check that professsional card Upgrade button opens purchase modal
+        // * Check that professsional card Upgrade button opens purchase modal
         cy.get('#pricingModal').get('#professional').get('#professional_action').click();
-        cy.get('.PurchaseModal').should('exist');
+        cy.get('.modal-body').should('exist');
```
Comment (nickmisasi) [inline e2e/cypress/tests/integration/enterprise/cloud/billing/cloud_pricing_modal_spec.js:231]: This class name is used by all modals in Mattermost - you may want to use something a little more specific for the longevity of the test.

**Review by nickmisasi — APPROVED** [review]
One small suggestion but otherwise LGTM - thanks @yasserfaraazkhan

File: e2e/cypress/tests/integration/enterprise/cloud/billing/cloud_pricing_modal_spec.js (hunk 1 above)
Comment (yasserfaraazkhan) [reply e2e/cypress/tests/integration/enterprise/cloud/billing/cloud_pricing_modal_spec.js:231]: ohh!! Thank you @nickmisasi ,
I'll test with another locator

**Review by furqanmlk — COMMENTED** [review]
Failed in locally
This method is forbidden

image

**Review by furqanmlk — APPROVED** [review]
Thanks @yasserfaraazkhan 
Tests are passing 🎉 

image

**Pull request merged**